### ISSUE TITLE
strict-comparisons rule complains about comparisons, it should allow

### ISSUE THREAD

**Issue by MaceWindu** [palantir/tslint#4780]
### Bug Report

- __TSLint version__: 5.18.0
- __TypeScript version__: 3.5.2
- __Running TSLint via__: gulp-tslint

#### TypeScript code being linted

```ts
public static test1(param?: boolean): boolean {
    return param === undefined ? false : param;
}
```

with `tslint.json` configuration:

```json
{
    "extends": [],
    "rules": {
        "strict-comparisons": true
    }
}
```

#### Actual behavior
```
ERROR: 31:20  strict-comparisons  Cannot use '===' comparator for type 'undefined'.
ERROR: 35:20  strict-comparisons  Cannot use '===' comparator for type 'undefined'.
ERROR: 39:20  strict-comparisons  Cannot use '===' comparator for type 'undefined'.
ERROR: 43:20  strict-comparisons  Cannot use '===' comparator for type 'null'.
ERROR: 47:20  strict-comparisons  Cannot use '===' comparator for type 'null'.
ERROR: 51:20  strict-comparisons  Cannot use '===' comparator for type 'null'.
```

#### Expected behavior
no errors

**Comment by MaceWindu** [comment]
Duplicate of #4774 
Didn't found at first due to incorrect rule name
